pkg/webhook/webhook.go

```diff
@@ -0,0 +1,80 @@
+package webhook
+
+import (
+	"github.com/kubeflow/kfserving/pkg/apis/serving"
+	"github.com/kubeflow/kfserving/pkg/apis/serving/v1alpha1"
+	"github.com/kubeflow/kfserving/pkg/webhook/admission/kfservice"
+	"k8s.io/api/admissionregistration/v1beta1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+	webooktypes "sigs.k8s.io/controller-runtime/pkg/webhook/types"
+)
+
+var (
+	// KFServingWebhookServerName is the name for the webhook server for KFServing resources
+	KFServingWebhookServerName = "kfserving-webhook-server"
+
+	// KFServingWebhookServerNamespace is the namespace for the webhook server for KFServing resources
+	KFServingWebhookServerNamespace = "kubeflow-system"
+)
+var log = logf.Log.WithName(KFServingWebhookServerName)
+
+// AddToManager adds all Controllers to the Manager
+// +kubebuilder:rbac:groups=admissionregistration.k8s.io,resources=mutatingwebhookconfigurations;validatingwebhookconfigurations,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups="",resources=secrets,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups="",resources=services,verbs=get;list;watch;create;update;patch;delete
+func AddToManager(manager manager.Manager) error {
+	server, err := webhook.NewServer("webhook-server", manager, webhook.ServerOptions{
+		Port:    9876,
+		CertDir: "/tmp/cert",
+		BootstrapOptions: &webhook.BootstrapOptions{
```

Comment:
Your masters need to call your goland (localhost), so it will vary depending on where your masters are (i.e. minikube/gke/xyz).

Can you add this to the developer guide?